packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/views/explorer/services/DeleteDocumentTreeItemEventHandler.java

```diff
@@ -38,31 +40,46 @@
 @Service
 public class DeleteDocumentTreeItemEventHandler implements IDeleteTreeItemHandler {
 
+    private final IRepresentationDescriptionSearchService representationDescriptionSearchService;
+
+    public DeleteDocumentTreeItemEventHandler(IRepresentationDescriptionSearchService representationDescriptionSearchService) {
+        this.representationDescriptionSearchService = Objects.requireNonNull(representationDescriptionSearchService);
+    }
+
     @Override
     public boolean canHandle(IEditingContext editingContext, TreeItem treeItem) {
         return treeItem.getKind().equals(ExplorerDescriptionProvider.DOCUMENT_KIND);
     }
 
     @Override
-    public IStatus handle(IEditingContext editingContext, TreeItem treeItem) {
+    public IStatus handle(IEditingContext editingContext, TreeItem treeItem, Tree tree) {
+        var optionalTreeDescription = this.getTreeDescription(editingContext, tree);
         var optionalEditingDomain = Optional.of(editingContext)
                 .filter(IEMFEditingContext.class::isInstance)
                 .map(IEMFEditingContext.class::cast)
                 .map(IEMFEditingContext::getDomain);
 
-        if (optionalEditingDomain.isPresent()) {
-            var editingDomain = optionalEditingDomain.get();
-            ResourceSet resourceSet = editingDomain.getResourceSet();
-            URI uri = new JSONResourceFactory().createResourceURI(treeItem.getId());
-
-            List<Resource> resourcesToDelete = resourceSet.getResources().stream()
-                    .filter(resource -> resource.getURI().equals(uri))
-                    .toList();
-            resourcesToDelete.forEach(resourceSet.getResources()::remove);
-
-            return new Success(ChangeKind.SEMANTIC_CHANGE, Map.of());
+        if (optionalEditingDomain.isPresent() && optionalTreeDescription.isPresent()) {
+            var variableManager = new VariableManager();
+            variableManager.put(IEditingContext.EDITING_CONTEXT, editingContext);
+            variableManager.put(TreeDescription.ID, treeItem.getId());
+            var object = optionalTreeDescription.get().getTreeItemObjectProvider().apply(variableManager);
+            if (object instanceof Resource resource) {
+                ResourceSet resourceSet = optionalEditingDomain.get().getResourceSet();
+                resourceSet.getResources().remove(resource);
+                return new Success(ChangeKind.SEMANTIC_CHANGE, Map.of());
+            }
         }
         return new Failure("");
     }
+
+    private Optional<TreeDescription> getTreeDescription(IEditingContext editingContext, Tree tree) {
+        return this.representationDescriptionSearchService
+        .findAll(editingContext).values().stream()
+        .filter(TreeDescription.class::isInstance)
+        .map(TreeDescription.class::cast)
+        .filter(td -> Objects.equals(td.getId(), tree.getDescriptionId()))
+        .findFirst();
+    }
```

Comment:
Can be removed and replaced inline with a simple call to `IRepresentationDescriptionSearchService#findById` instead

Reply:
Done